services/py-api/tests/integration_tests/conftest.py

```diff
@@ -1,15 +1,84 @@
+# https://pytest-asyncio.readthedocs.io/en/latest/concepts.html
+# https://pytest-asyncio.readthedocs.io/en/latest/how-to-guides/run_session_tests_in_same_loop.html
+# https://docs.pytest.org/en/stable/how-to/writing_hook_functions.html
+def pytest_collection_modifyitems(items: List[pytest.Item]) -> None:
+    pytest_asyncio_tests = (item for item in items if pytest_asyncio.is_async_test(item))
+    session_scope_marker = pytest.mark.asyncio(loop_scope="session")
+    for async_test in pytest_asyncio_tests:
+        async_test.add_marker(session_scope_marker, append=False)
+
+
+# The scope is set to session so that that Async Client is only initialized once throughout the testing session,
+# instead of initializing it on every test function invocation
+# Read More: https://docs.pytest.org/en/stable/how-to/fixtures.html#scope-sharing-fixtures-across-classes-modules-packages-or-session
+@pytest_asyncio.fixture(scope="session")
 async def async_client() -> AsyncClient:
+    LOG.debug("Opening Async Client")
     client = AsyncClient(transport=ASGITransport(app=app), base_url="http://test")
     yield client
     LOG.debug("Closing Async Client")
     await client.aclose()
+
+
+@patch.dict(environ, {"SECRET_AUTH_TOKEN": "OFFLINE_TOKEN"})
+async def clean_up_test_participant(async_client: AsyncClient, result_json: Dict[str, Any]) -> None:
+
+    participant_id = result_json["participant"]["id"]
+    await async_client.delete(
+        url=f"{PARTICIPANT_ENDPOINT_URL}/{participant_id}",
+        headers={"Authorization": f"Bearer {environ['SECRET_AUTH_TOKEN']}"},
+    )
+
+    if result_json["team"] is not None:
+        team_id = result_json["team"]["id"]
+        await async_client.delete(
+            url=f"{TEAM_ENDPOINT_URL}/{team_id}",
+            headers={"Authorization": f"Bearer {environ['SECRET_AUTH_TOKEN']}"},
+        )
+
+
+# The following is an exapmle of factories as fixtures in pytest
+# It manages the creation of participants and ensures the cleanup process after every test function
+# You can read more about that: https://docs.pytest.org/en/stable/how-to/fixtures.html#factories-as-fixtures
+# It uses the same philosophy for the teardown as it is suggested on the example of the docs above.
+# Learn more about fixture teardown here: https://docs.pytest.org/en/stable/how-to/fixtures.html#teardown-cleanup-aka-fixture-finalization
+@pytest_asyncio.fixture
+async def create_test_participant(async_client: AsyncClient) -> Callable[..., Dict[str, Any]]:
+
+    request_results = []
+
+    async def _create(participant_body: Dict[str, Any]) -> Response:
+
+        LOG.debug("Creating a test participant")
+        result = await async_client.post(PARTICIPANT_ENDPOINT_URL, json=participant_body)
+        request_results.append(result)
+        return result
+
+    yield _create
+
+    LOG.debug("Cleaning up the test participants that were successfully created")
```

Comment:
maybe move the log inside the for loop of the if. Because the log again is misleading:
<img width="1333" alt="image" src="https://github.com/user-attachments/assets/e4107f03-5c08-4608-844b-9c116e7a3466" />
Here, a participant is not inserted, and we are still "removing them". (the log from the conftest and the partcipant_repository)

Reply:
Fixed in f9eb4a0865daa13b7c12760f2b66c73f3746d300